### PR TITLE
Crewmonitor fixes and refactors

### PR DESCRIPTION
**Pull request by Mat05usz**
## About The Pull Request

Ports: https://github.com/tgstation/tgstation/pull/55124

while also applying my own changes to make it compatible with our virtual z level stuff.

What this PR basically is a bunch of cleanup in crew console code and, what's more important,
fixes custom jobs appearing as bolded and at the top of crew console list (they've never been intended to appear there, only with other 'civilians') and Unknowns are no longer bolded with a weird '()' next to them.

Tested it on a bunch of Z-levels, virtual or not, on MultiZ Debug, with people in/out closets etc. I'd say it's working fine in any scenario I could think of, but it probably sucks anyways.

Finally, as stated in the original PR, moves sorting of the list to client-side, because there is no reason to keep it on the server.

## Why It's Good For The Game

Crew console working and displaying jobs as intended.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

**On the same z-level:**
image

**On different z-levels:**
image

**Same virtual z-level:**
image

**Another virtual z-level:**
image

</details>

## Changelog
:cl: bobbahbrown, Mat05usz
fix: Crew console is properly displaying custom jobs and unknowns
refactor: Cleans up crew console code
/:cl:

### CODE REVIEW

**Comment by Mat05usz** [comment]
Regarding my previous failed PR, turns out that when I hastly set my global email in github config to 'aaa@bbb.com' some dude named xkarlois had it, and it signed all commits as him lmao. Hope that noone being an owner of commits doesn't break stuff.

File: code/game/machinery/computer/crew.dm [hunk 1]
```diff
@@ -21,64 +25,78 @@
...
+		"Bartender" = 61,
+		"Cook" = 62,
+		"Botanist" = 63,
+		"Curator" = 64,
+		"Chaplain" = 65,
+		"Clown" = 66,
+		"Mime" = 67,
+		"Janitor" = 68,
+		"Lawyer" = 69,
+		"Barber" = 70,
```
Comment (chloecat0) [inline code/game/machinery/computer/crew.dm]: Won't barbers be considered a head of staff because it's divisible by 10?

Comment (Mat05usz) [reply code/game/machinery/computer/crew.dm]: You are correct, they will. Thanks for pointing that out!

**Comment by PowerfulBacon** [comment]
Mainly looked at the virtual z-level stuff, anything related to that seems fine to me

**Review by Archanial — APPROVED** [review]

**Comment by Dejaku51** [comment]
You broke crew monitors

**Pull request merged**